src/Microsoft.AspNetCore.Razor.Language/Legacy/TagHelperParseTreeRewriter.cs

```diff
@@ -511,7 +511,7 @@ private void ValidateParentAllowsPlainTag(Block tagBlock, ErrorSink errorSink)
             var currentTracker = _trackerStack.Count > 0 ? _trackerStack.Peek() : null;
 
             if (HasAllowedChildren() &&
-                !_currentTagHelperTracker.AllowedChildren.Contains(tagName, StringComparer.OrdinalIgnoreCase))
+                !_currentTagHelperTracker.PrefixedAllowedChildren.Contains(tagName, StringComparer.OrdinalIgnoreCase))
```

Comment:
This is not part of this fix. But when I looked at it I didn't understand why we are comparing with `AllowedChildren` instead of `PrefixedAllowedChildren`. Just wanted to bring this up.